core/src/main/java/dk/alexandra/fresco/framework/network/NetworkPerformanceDelegate.java

```diff
@@ -0,0 +1,91 @@
+package dk.alexandra.fresco.framework.network;
+
+import dk.alexandra.fresco.framework.PerformanceLogger;
+import dk.alexandra.fresco.framework.configuration.NetworkConfiguration;
+import dk.alexandra.fresco.framework.util.Pair;
+import java.io.IOException;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+public class NetworkPerformanceDelegate extends PerformanceLogger implements Network {
+
+  private Network network;
```

Comment:
My previous comment was apparently confusing.

The class should be called NetworkLoggingDecorator. 
The inner field decorated or delegate.

The inheritance is currently used for passing the partyid into the logging. Change the PerformanceLogger to an interface and let print tak the party as an argument? Or a decorated logger.

Reply:
Yea okay - I was indeed confused :) - will do